helmfile/charts/grafana-dashboards/templates/configmap-dashboards.yaml

```diff
@@ -1,31 +1,35 @@
-{{- range $key, $value := .Values.dashboards }}
-{{- $_ := set $ "key" $key }}
-{{- $_ := set $ "value" $value }}
-{{- if $value.enabled }}
+{{- $files := .Files.Glob "dashboards/**dashboard.json" }}
+{{- range $path, $fileContents := $files }}
+{{- $dashboardName := regexReplaceAll "(^.*/)(.*)\\.json$" $path "${2}" }}
+{{- if not (has $dashboardName $.Values.disabledDashboards) }}
 ---
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: {{ printf "%s-%s" $.Release.Name $key }}
-  namespace: {{ $.Release.Namespace }}
+  name: {{ printf "%s-%s" (include "grafana-dashboards.fullname" $) $dashboardName | trunc 63 | trimSuffix "-" }}
+  namespace: {{ include "grafana-dashboards.namespace" $ }}
   labels:
-    {{- if $value.user_visible }}
-    {{ $.Values.labelKey }}: "1"
-    {{- else }}
+    {{- include "grafana-dashboards.labels" $ | nindent 4 }}
+    {{- if has $dashboardName $.Values.notDeveloperVisible }}
     {{ $.Values.labelKey }}: "ops"
+    {{- else }}
+    {{ $.Values.labelKey }}: "1"
     {{- end }}
 data:
-  {{- if eq $key "backup" }}
-  {{- $sync := (empty $value.sync) | ternary "**disabled**" (printf "**enabled** for bucket(s) %s" (join ", " (sortAlpha $value.sync))) }}
-  {{ $key }}.json: |-
-    {{- regexReplaceAll "<<rclone-sync-state>>" ($.Files.Get (printf "dashboards/%s-dashboard.json" $key)) $sync | nindent 6 }}
-  {{- else if eq $key "welcome" }}
+  {{- if eq $dashboardName "backup-dashboard" }}
+  {{- $sync := (empty $.Values.sync) | ternary "**disabled**" (printf "**enabled** for bucket(s) %s" (join ", " (sortAlpha $.Values.sync))) }}
+  {{ $dashboardName }}.json: |-
```

Comment:
Small nit, but couldn't this line be moved outside of the `{{- if eq ...` since they are all the same?
I'm not sure if it would be more readable though so I'm not convinced myself. What are your though?

Reply:
to me it looked easier to read if we have them
I pushed a commit with you suggestion anyway

Reply:
After seeing both I don't really have a preference. So pick the one you think is best.